[GSgiansen] iP

GSgiansen opened the pull request:
# MeowBot

> "Your cat's mind is for having ideas, not holding them" - Gian Sen

MeowBot frees your mind by having your imaginary cat remember things for you. It's

- text-based to make you feel like hackerman
- easy to learn
- ~~slow~~ *SUPER FAST* to use
- who doesnt like cats ? 

All you need to do is,

1. imaginge you have a cat
2. download it from [here](https://github.com/GSgiansen/ip/releases/tag/A-Jar)
3. add your tasks
4. let your imaginary cat manage your tasks for you :cat:

What's best it is **FREEEEEEEEE**!

## Features:

- [x] Free virtual cat
- [x] Lists tasks with date
- [x] Add delight to the experience when all tasks are complete :tada:
- [ ] Managing deadline (Coming soon)

If youre are a gigachad JAVA programmer, you can use it to practice Java too! Here's the `main` method:

```java
    public class Main {
        public static void main(String[] args) throws DukeException, IOException {
            new MeowBot("src/main/data/meowbot.txt").run();
        }
    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

GSgiansen committed:
Completed Level-0 features for MeowBot

GSgiansen committed:
Completed Level-1 features that Echos whatever the user inputs and exits when bye is executed

GSgiansen committed:
Completed Level-2 features where users can add tasks and view task list

GSgiansen committed:
Fixed list indexing to show from 1 onwards instead of 0

GSgiansen committed:
Added ability to mark and unmark tasks and display them

GSgiansen committed:
Completed Level-4 features which includes string processing and displaying of 3 diff types of events

GSgiansen committed:
Added semi-automatic testing of the text prompts

GSgiansen committed:
Added DukeException to handle mistyped keyword or if the todo description is left blank

GSgiansen committed:
Added DELETE functionality to MeowBot

GSgiansen committed:
Added more encapsulation to tidy up and refactor code

GSgiansen committed:
Change main file name to MeowBot

GSgiansen committed:
Add saving tasks to local storage for level-7 Criteria

GSgiansen committed:
Merge branch 'branch-level-7'

GSgiansen committed:
ChatBot is now able to read date time and process them in readable huan word form

GSgiansen committed:
Change DukeException throwing

GSgiansen committed:
Changed Duke Exception handling

GSgiansen committed:
Merge branch 'branch-Level-8'

GSgiansen committed:
Abstract out Storage,Command,Parser,Ui

GSgiansen committed:
Add meowbot.txt for local storage

GSgiansen committed:
Merge branch 'A-MoreOOP' for additional OOP practics

GSgiansen committed:
MeowBot.txt edited

GSgiansen committed:
Merge tag 'A-MoreOOP'

A-MoreOOP as more OOP practices has been adopted

GSgiansen committed:
Revise git control

GSgiansen committed:
Merge branch 'A-MoreOOP'

GSgiansen committed:
Package files to respective packages

GSgiansen committed:
Edit filepaths for the different packages

GSgiansen committed:
Merge branch 'A-Packages' to packegerize code

GSgiansen committed:
Merge branch 'add-gradle-support' into A-Gradle

GSgiansen committed:
Merge branch 'A-Gradle' into master branch

GSgiansen committed:
Add checks for Event date validity and Parser command validity

GSgiansen committed:
Testcases for Events task and Parser class

GSgiansen committed:
Merge branch 'A-JUnit' into master with additional testcases

GSgiansen committed:
Add JAR file for download for new users

GSgiansen committed:
Merge branch 'A-Jar' into the master branch

GSgiansen committed:
Add Header comments for the classes

GSgiansen committed:
Merge branch 'branch-A-Javadoc' into main

GSgiansen committed:
Fix whitespaces and coding standard

GSgiansen committed:
Merge branch 'branch-A-CodingStandard' into master

GSgiansen committed:
Add find command utility to find tasks with relevant keywords

GSgiansen committed:
Add FindCommand class and utility

GSgiansen committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/task/TaskList.java

GSgiansen committed:
ChatBot has functionality to find tasks with keyword

GSgiansen committed:
Added header comment to AddCommand

GSgiansen committed:
Fix Coding standard for AddCommand, MeowBot, Storage class

GSgiansen committed:
Fix coding standards for all classes

GSgiansen committed:
Merge branch 'branch-A-CodingStandard' into master with improved coding standard

GSgiansen committed:
Abstract print statements to Ui class

GSgiansen committed:
Merge branch 'branch-A-MoreOOP' and abstracting print statements to Ui class

GSgiansen committed:
Add checkStyle static analysis

GSgiansen committed:
Merge branch 'branch-A-CheckStyle' into master to ensure style

GSgiansen committed:
Add UserGUI with coloured background

GSgiansen committed:
Add GUI content and allow meow bot to respond to user input and output to the GUI

GSgiansen committed:
Fix zero indexing for tasks in TaskList class

GSgiansen committed:
Fix issue with MeowBot when encountered erro

GSgiansen committed:
Update MeowBot profile picture and adjust error message

GSgiansen committed:
Fix new line for command

GSgiansen committed:
Stylecheck fix

GSgiansen committed:
Merge branch 'branch-Level-10' into master

Mahidharah commented on src/main/java/duke/MeowBot.java:
```diff
@@ -0,0 +1,126 @@
+     * @throws DukeException when generating the tasks back from the data file
+     * @throws IOException when the data file cannot be found
+     */
+
```
Unnecessary line here between javadoc comment and method. Consider removing this to keep thinks more compact!

Mahidharah commented on src/main/java/duke/helper/Storage.java:
```diff
@@ -0,0 +1,128 @@
+/**
+ * Storage is the local class that deals with the logic for storing and updating tasks data locally
+ */
+
```
Very large space here! Consider cutting down on it to make things more readable.

Mahidharah commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+    public void setType(String type) {
+        this.type = type;
+    }
+    public void setOgName(String ogName) {
```
Peculiar naming... I dont know what Og is. Consider changing it to something more obvious

Mahidharah reviewed (commented):
Just some nits to fix, but overall really well done with the commands and all that. Neat Code!

GSgiansen committed:
Update gradle to build new JAR release

GSgiansen committed:
Add assertions at key point of code to ensure that the build functions per assumption

GSgiansen committed:
Enhance code quality and remove redundant code

GSgiansen committed:
Review comments from PR

GSgiansen committed:
Merge pull request #1 from GSgiansen/branch-A-CodeQuality

Branch A-CodeQuality

GSgiansen committed:
Resolve merge conflict between master and CodingQuality

GSgiansen committed:
Merge pull request #2 from GSgiansen/branch-A-Assertions

Branch A-Assertions

GSgiansen committed:
MeowBot has improved functionality

MeowBot now has its own GUI to visualise the conversation better!

MeowBot now supports the following functions:
"bye", "list", "unmark", "mark", "todo", "deadline",
"event", "delete", "find"

GSgiansen committed:
Add simple CI file

GSgiansen committed:
Merge pull request #3 from GSgiansen/branch-A-CI

Add simple CI file

GSgiansen committed:
Add duplicate detection functionality

GSgiansen committed:
DukeException is thrown when duplicate task observed

GSgiansen committed:
Unable to mark/delete/unmark tasks outside of range

GSgiansen committed:
Merge pull request #4 from GSgiansen/branch-C-DetectDuplicates

Branch C Detect Duplicates completed

GSgiansen committed:
Fix CheckStyle for MeowBot

GSgiansen committed:
Fetch from repo

GSgiansen committed:
Merge branch 'branch-C-DetectDuplicates'

GSgiansen committed:
Edit build.gradle

GSgiansen committed:
Remove main class setting unneccessarily

GSgiansen committed:
Remove IO tests as GUI is present

GSgiansen committed:
Add additional testing for comprehensiveness

GSgiansen committed:
Fix StyleCheck

GSgiansen committed:
Additional testing to ensure product works

GSgiansen committed:
Update readme to include user guide

GSgiansen committed:
Merge branch 'branch-A-UserGuide'

GSgiansen committed:
Add Ui.png

GSgiansen committed:
Generate fat JAR file

GSgiansen committed:
Create JAR file to run

GSgiansen committed:
Edit website header

GSgiansen committed:
Edit website and Ui pic

GSgiansen committed:
Edit ip website header

GSgiansen committed:
Modify Pages README

GSgiansen committed:
Update README.md

GSgiansen committed:
Fix CI/CD
